src/Noise/InterpolNoise.h

```diff
@@ -427,28 +427,37 @@ class cInterpolNoise
 
 		// Calculate query values using Cell:
 		int FromZ = 0;
-		for (int z = 0; z < NumSameZ; z++)
+		for (int z = 0; z < NumSameZ;)
 		{
 			int ToZ = FromZ + SameZ[z];
 			int CurFloorZ = FloorZ[FromZ];
 			int FromY = 0;
-			for (int y = 0; y < NumSameY; y++)
+			for (int y = 0; y < NumSameY;)
 			{
 				int ToY = FromY + SameY[y];
 				int CurFloorY = FloorY[FromY];
 				int FromX = 0;
-				for (int x = 0; x < NumSameX; x++)
+				for (int x = 0; x < NumSameX;)
 				{
 					int ToX = FromX + SameX[x];
 					Cell.Generate(FromX, ToX, FromY, ToY, FromZ, ToZ);
-					Cell.Move(FloorX[ToX], CurFloorY, CurFloorZ);
-					FromX = ToX;
+					if (++x < NumSameX)
```

Comment:
```suggestion
					if (++x < NumSameX)
					if (++x < NumSameX)  // Call Move() everytime except for the last loop iteration
```
Etc. in respective locations